docs/rest-api-fundamentals/README.md

```diff
@@ -0,0 +1,169 @@
+| 429    | Too Many Requests      | The user has too many outstanding requests.   See [Throttling](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#14-throttling-quotas-and-limits)                     |
+| 500    | Internal Server Error  | An unexpected internal error occurred and the request was not processed.                                                                                                                 |
+| 503    | Service Unavailable    | The service is unable to perform the request due to general load or other problems outside the control of the individual caller.                                                         |
+
+> To reduce load on the system, services should be designed so that returning 429 or 503 is as inexpensive as possible, either by putting in special fastpath code, or ideally by depending on a common frontdoor or load balancer that provides this functionality.
+
+## URL Structure
+
+### Overview
+
+Humans should be able to easily read and construct URLs.
+
+An example of a well-structured URL is:
+
+```http
+https://api.contoso.com/v1.0/people/jdoe@contoso.com/inbox
+```
+
+An example URL that is not friendly is:
+
+```http
+https://api.contoso.com/EWS/OData/Users('jdoe@microsoft.com')/Folders('AAMkADdiYzI1MjUzLTk4MjQtNDQ1Yy05YjJkLWNlMzMzYmIzNTY0MwAuAAAAAACzMsPHYH6HQoSwfdpDx-2bAQCXhUk6PC1dS7AERFluCgBfAAABo58UAAA=')
+```
+
+### Collections
+
+Collections are located directly under the service root when they are top level, or as a segment under another resource when scoped to that resource.
+
+For example:
+
+```http
+GET https://api.contoso.com/v1.0/articles
+```
+
+Whenever possible, services MUST support the "/" pattern.
+For example:
+
+```http
+GET https://{serviceRoot}/{collection}/{id}
+```
+
+Where:
+
+* {serviceRoot} – the combination of host (site URL) + the root path to the service
+* {collection} – the name of the collection, unabbreviated, pluralized
+* {id} – the value of the unique id property. When using the "/" pattern this MUST be the raw string/number/guid value with no quoting but properly escaped to fit in a URL segment.
+
+### Nested collections
+
+Collection items may contain other collections.
+For example, a user collection may contain user resources that have multiple addresses:
+
+```http
+GET https://api.contoso.com/v1.0/articles/123/comments
+```
+
+```json
+{
+  "value": [
+    { "id": 3, "body": "1st comment", "author_id": 321 },
+    { "id": 8, "body": "Next comment", "author_id": 456 }
+  ]
+}
+```
+
+While technically one could imagine scoping even further with an endpoint such as `GET https://api.contoso.com/v1.0/articles/123/comments/3/author`, practically this can become quite cumbersome to manage. Limit nested collections to only one or two collections deep.  In this example `/authors` is likely a top level collection that can now be intuitively retrieved by using the provided `author_id` via `GET https://api.contoso.com/v1.0/authors/321`.
```

Comment:
```suggestion
While technically one could imagine scoping even further with an endpoint such as `GET https://api.contoso.com/v1.0/articles/123/comments/3/author`, practically this can become quite cumbersome to manage. Limit nested collections to only one or two collections deep. In this example `/authors` is likely a top level collection that can now be intuitively retrieved by using the provided `author_id` via `GET https://api.contoso.com/v1.0/authors/321`.
```